pkg/infra/httpclient/httpclientprovider/http_client_provider.go

```diff
@@ -0,0 +1,30 @@
+package httpclientprovider
+
+import (
+	"fmt"
+
+	sdkhttpclient "github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+	"github.com/grafana/grafana/pkg/infra/httpclient"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+var newProviderFunc = sdkhttpclient.NewProvider
+
+// New creates a new HTTP client provider with pre-configured middlewares.
+func New(cfg *setting.Cfg) httpclient.Provider {
+	userAgent := fmt.Sprintf("Grafana/%s", cfg.BuildVersion)
+	middlewares := []sdkhttpclient.Middleware{
+		DataSourceMetricsMiddleware(),
+		SetUserAgentMiddleware(userAgent),
+		sdkhttpclient.BasicAuthenticationMiddleware(),
+		sdkhttpclient.CustomHeadersMiddleware(),
+	}
+
+	if cfg.SigV4AuthEnabled {
+		middlewares = append(middlewares, SigV4Middleware())
```

Comment:
It looks this way but just to confirm, this will mean SigV4 will always be invoked last in the middleware chain right?

Reply:
Yes. That's how it is right now and what we want right?

Reply:
Exactly 👍